### PR TITLE
fix: telemetry failure should not cause CLI failure

### PR DESCRIPTION
**Pull request by anshbansal**
## Checklist
- [x] The PR conforms to DataHub's [Contributing Guideline](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable)

### CODE REVIEW

File: metadata-ingestion/src/datahub/telemetry/telemetry.py [hunk 1]
```diff
@@ -123,14 +123,17 @@ def init_tracking(self) -> None:
...
+            self.mp.people_set(
+                self.client_id,
+                {
+                    "datahub_version": datahub_package.nice_version_name(),
+                    "os": platform.system(),
+                    "python_version": platform.python_version(),
+                },
+            )
+        except Exception as e:
+            logger.debug(f"Error reporting telemetry: {e}")
```
Comment (rslanka) [inline metadata-ingestion/src/datahub/telemetry/telemetry.py:136]: Shouldn't this should be at warning level?

Comment (anshbansal) [reply metadata-ingestion/src/datahub/telemetry/telemetry.py]: fixed

File: metadata-ingestion/src/datahub/telemetry/telemetry.py [hunk 2]
```diff
@@ -123,14 +123,17 @@ def init_tracking(self) -> None:
...
+            self.mp.people_set(
+                self.client_id,
+                {
+                    "datahub_version": datahub_package.nice_version_name(),
+                    "os": platform.system(),
+                    "python_version": platform.python_version(),
+                },
+            )
+        except Exception as e:
+            logger.warn(f"Error reporting telemetry: {e}")
```
Comment (rslanka) [inline metadata-ingestion/src/datahub/telemetry/telemetry.py:136]: logger.warn is deprecated since 3.3. Please use logger.warning.

Comment (rslanka) [inline metadata-ingestion/src/datahub/telemetry/telemetry.py:136]:
```suggestion
            logger.warning(f"Error reporting telemetry: {e}")
```

Comment (anshbansal) [reply metadata-ingestion/src/datahub/telemetry/telemetry.py]: fixed

**Pull request merged**